File: services/order-service/src/main/java/io/minishop/order/web/dto/OrderItemResponse.java
package io.minishop.order.web.dto;

import io.minishop.order.domain.OrderItem;

import java.math.BigDecimal;

public record OrderItemResponse(
        Long id,
        Long productId,
        Integer quantity,
        BigDecimal price
) {
    public static OrderItemResponse from(OrderItem item) {
        return new OrderItemResponse(item.getId(), item.getProductId(), item.getQuantity(), item.getPrice());
    }
}

File: services/order-service/src/main/kotlin/io/minishop/order/web/dto/OrderItemResponse.kt
package io.minishop.order.web.dto

import io.minishop.order.domain.OrderItem
import java.math.BigDecimal

@JvmRecord
data class OrderItemResponse(
    val id: Long?,
    val productId: Long?,
    val quantity: Int?,
    val price: BigDecimal?,
) {
    companion object {
        @JvmStatic
        fun from(item: OrderItem): OrderItemResponse =
            OrderItemResponse(item.id, item.productId, item.quantity, item.price)
    }
}
